Non-Interactive-Protocol.md

```diff
@@ -0,0 +1,212 @@
+expiration.
+
+# General Requirements
+
+A DLC client should have censorship-resistant access to the blockchain, use a
+local fee-estimator and ensure it's well-connected to the tx-relay network.
+
+We recommend to not react on any mempool event as otherwise it would be a source of
+cross-layer mapping. <sup>[Cross-layer Mapping](https://arxiv.org/pdf/2007.00764.pdf)</sup>
+
+We recommend to not produce local non-adaptor signatures for transaction until their broadcast
+is required to avoid their misuage in case of site compromise, assuming an external signer.
+
+Further, we recommend duplicate backups of counterparty signatures to avoid relying on
+counterparty honest behavior in case of primary storage failure.
+
+# Fee bumping policy
+
+A onchain state machine must take actions based on different timers parameterized by two
+values: `congestion_delay`/`security_delay`. Their definition and setting recommendations
+are laid out in this section, their usage is precised in following sections.
+
+Client's `congestion_delay`/`security_delay` can be scheduled on three different basis:
+- a block height
+- a local clock
+- an observed feerate fluctuation
+
+Note, these criterias might be observable by an adversary and we recommend to implementations
+to pad any criteria with a random delay to diminish risks of tx-relay jamming.
+
+We don't provide recommendations for the `congestion delay` as its settings depends of user
+liquidity preferences and available local data to predict the feerate. Confirmation of non
+time-sensitive transactions should happen but a user may prefer to delay them in case of
+mempool-congestion.
+
+We provide recommendations for the `security_delay` as any transaction under this fee-bumping
+policy has a time-sensitive confirmation. A delay of 1 block and a fast-confirmation feerate
+as reported by fee-estimator should be preferred.
+
+# Funding Phase
+
+A party always starts the protocol in the funding phase. The funding phase can
+terminate by either a transition to the execution phase or a protocol abort.
+
+## Requirements
+
+A node:
+
+  - if a counterparty's funding signature is received:
+    - MUST finalize the funding transaction with a local `SIGHASH_ALL` signatures
+    - MUST broadcast the funding transaction
+
+A node:
+
+  - if a funding signature has been sent or a funding transaction broadcast and there is no confirmation after 6 blocks:
+    - MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+    - MUST start a fee-bumping timer of length `congestion_delay`
+    - if `congestion_delay` expires:
+      - MUST spend the change output with a CPFP increasing package feerate to shorten confirmation time
+      - MUST replace the CPFP transaction according to BIP 125
+      - MUST reschedule timer for next bump
+    - if `contract_maturity_bound` is at least 20 blocks ahead from local tip:
```

Comment:
If you want a suggestion I would be more comfortable with something like `(contract_maturity_bound + signing_time)/2`. I really think that using something like 20 provides absolutely no value here because I would expect most `contract_maturity_bound`s in the world to have a margin of error well over 20 blocks.

Reply:
And on a separate note, I am also concerned about what this could mean for events which an oracle will either sign at any time present to contract maturity (such as an event like "will btc price exceed 20k before end of year" which could generate a "yes" signature at any time but will only generate a "no" signature at new year). In this case it seems like some number of blocks after signing (or worse, if a price movement is happening) then parties will want to enter this state

Reply:
And also, another issue with using `contract_maturity_bound` as closely as 20 is that I see virtually no difference in the incentive to double-spend a long-standing DLC 20 blocks before contract maturity and at contract maturity, either way the counter-party likely has near certainty about what the outcome is going to be which they did not have when they agreed to the DLC

Reply:
> And also, another issue with using contract_maturity_bound as closely as 20 is that I see virtually no difference in the incentive to double-spend a long-standing DLC 20 blocks before contract maturity

In this case, it's not about counterparty incentive, but your block delay expectation to confirm your double-spend. I was thinking 3 blocks (fast-confirmation) plus 17 of security margin, based on roughly what we're using to accelerate bumping timer in RL.

But your point about anytime-signing event is a really good one, and added with the general observation that we don't have a way to make signing time a block-or-epoch consensus-enforced fixed point, we should defer more to clients.

What about introducing a `security_point` to which the evaluation is left to client, with recommendations on how to define it according to event types (anytime-signing, long-standing, ...) but still with a block confirmation safety buffer (the current 20 blocks) and a fee-bumping timer requirement ?